### PR TITLE
Bug 1159230 - Remove old mozilla.org home page template

### PR DESCRIPTION
**Pull request by malena**
https://bugzilla.mozilla.org/show_bug.cgi?id=1159230

### CODE REVIEW

**Comment by sgarrity** [comment]
I *think* the [failed test](https://travis-ci.org/mozilla/bedrock/builds/63377548) here is due to a test that has been rendered out-of-date (or obsolete) by this PR.

This test is referencing `mozorg/home.html`, which is no longer used (and is removed by this PR): https://github.com/mozilla/bedrock/blob/master/lib/l10n_utils/tests/test_commands.py#L128

@flodolo Do you know if this test should be updated to point to the new home page template, or if it is still relevant?

**Comment by flodolo** [comment]
I'm not familiar at all with that test, I think we can just move it to a different "stable" template (maybe firefox/new?)

**Comment by malena** [comment]
Test are passing, ready for review.

File: bedrock/mozorg/views.py [hunk 1]
```diff
@@ -422,7 +422,7 @@ def new_home(request, locale=None):
 
 def home(request):
```
Comment (jpetto) [inline bedrock/mozorg/views.py]: Now that we only have one home page template, do we need these two different views? Do we still need the `HomeTestView` class above? It seems all code paths now lead to the same `home.html` template.

File: bedrock/settings/static_media.py [hunk 2]
```diff
@@ -583,41 +583,28 @@
...
         'source_filenames': (
-            'css/mozorg/home/home.less',
-            'css/mozorg/home/home-promo.less',
+            'css/mozorg/home/home-ie8.less',
         ),
-        'output_filename': 'css/home-2015-bundle.css',
+        'output_filename': 'css/home-ie8-bundle.css',
     },
-    'home-2015-ie8': {
+    'home-ie9': {
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: Is this bundle used anywhere?

File: bedrock/settings/static_media.py [hunk 3]
```diff
@@ -1399,13 +1377,13 @@
             'js/mozorg/home/ga-tracking.js',
             'js/mozorg/home/scroll-prompt.js',
         ),
-        'output_filename': 'js/home-2015-bundle.js',
+        'output_filename': 'js/home-bundle.js',
     },
-    'home-2015-ie9': {
+    'home-ie9': {
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: We have a `matchmedia_addlistener` bundle set up for these cases (where it's just the addListener shim needed). Better to use that bundle and remove this one.

File: bedrock/settings/static_media.py [hunk 4]
```diff
@@ -583,41 +583,28 @@
     },
     'home': {
         'source_filenames': (
-            'css/mozorg/home.less',
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: Should we get rid of the `*.less` and `*.js` files from the removed template?

File: bedrock/mozorg/views.py (hunk 1 above)
Comment (malena) [inline bedrock/mozorg/views.py]: @jpetto Thanks! I wasn't too sure about this. I will remove HomeTestView and just reference one view.

Comment (jpetto) [inline bedrock/mozorg/views.py]: I'm not 100% sure either. :smile: But as far as I can tell, there's no longer a reason to have 2 views and 1 class view. Should just need a single view. @jgmize - Care to weigh in?

If we can/do remove `HomeTestView`, some tests will need updating.

Comment (jgmize) [inline bedrock/mozorg/views.py]: Yes, I agree that we should consolidate to one view and update the tests as necessary.

**Comment by jpetto** [comment]
@malena - I think [this gist](https://gist.github.com/jpetto/5376bf49ff61c392449e) will help get the tests functional, though @jgmize may have a more elegant solution.

**Comment by malena** [comment]
@jpetto Thanks!! I've also made the rest of the changes you noted above.

File: bedrock/settings/static_media.py [hunk 5]
```diff
@@ -630,41 +630,22 @@
...
-        ),
-        'output_filename': 'css/home-ie9-bundle.css',
-    },
-    'home-ie8': {
-        'source_filenames': (
-            'css/mozorg/home-ie8.less',
-        ),
-        'output_filename': 'css/home-ie8-bundle.css',
-    },
     'home-ie': {
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: Is this bundle still needed? I don't see it included in any templates. If not, let's drop it and remove `home-ie.less`.

File: bedrock/settings/static_media.py [hunk 6]
```diff
@@ -1464,7 +1436,7 @@
         ),
         'output_filename': 'js/growth-firstrun-test2-bundle.js',
     },
-    'home-2015': {
+    'home': {
         'source_filenames': (
             'js/base/mozilla-share-cta.js',
             'js/base/firefox-anniversary-video.js',
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: Doesn't look like this video is on the home page anymore. Can probably remove this file from the bundle.

File: bedrock/settings/static_media.py [hunk 7]
```diff
@@ -1464,7 +1436,7 @@
         ),
         'output_filename': 'js/growth-firstrun-test2-bundle.js',
     },
-    'home-2015': {
+    'home': {
         'source_filenames': (
             'js/base/mozilla-share-cta.js',
```
Comment (jpetto) [inline bedrock/settings/static_media.py]: Hm, I don't see our `share_cta` macro on the home page either. Maybe remove this file also?

**Comment by malena** [comment]
@jpetto. Good eye! I assumed the stuff under home was current.

File: bedrock/mozorg/views.py [hunk 8]
```diff
@@ -493,20 +480,11 @@ def home_tweets(locale):
     return []
 
 
-def new_home(request, locale=None):
+def home(request, locale=None):
```
Comment (jpetto) [inline bedrock/mozorg/views.py]: We can now also drop the `locale` param from this function signature. Just set it to `locale = l10n_utils.get_locale(request)` below.

**Comment by jpetto** [comment]
One last comment on the `home` view definition. Can you squash the commits as well? Should be good after that.

Thanks!

**Comment by malena** [comment]
@jpetto All squashed. Thanks!

**Comment by jpetto** [comment]
Looks great @malena. Thanks!

r+ :zap:

**Comment by flodolo** [comment]
Adding tag to remember to clean up the old lang file.

**Pull request merged**